sdk/tests/conformance2/renderbuffers/readbuffer.html

```diff
@@ -0,0 +1,192 @@
+  wtu.glErrorShouldBe(gl, gl.INVALID_ENUM,
+      "calling readBuffer with GL_COLOR_ATTACHMENTi that exceeds MAX_COLOR_ATTACHMENT on fbo should generate INVALID_ENUM.");
+  gl.readBuffer(gl.COLOR_ATTACHMENT1);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "calling readBuffer with GL_COLOR_ATTACHMENT1 on the fbo should succeed.");
+  shouldBe('gl.getParameter(gl.READ_BUFFER)', 'gl.COLOR_ATTACHMENT1');
+
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION,
+      "should generate INVALID_OPERATION when reading from GL_COLOR_ATTACHMENT1 but this attachment has no image currently.");
+
+  var colorbuffer1 = gl.createRenderbuffer();
+  gl.bindRenderbuffer(gl.RENDERBUFFER, colorbuffer1);
+  gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, canvas.width, canvas.height);
+  gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT1, gl.RENDERBUFFER, colorbuffer1);
+  var fboColor1 = [0, 0, 255, 255];
+  gl.clearColor(fboColor1[0] / 255, fboColor1[1] / 255, fboColor1[2] / 255, fboColor[3] / 255);
+  gl.drawBuffers([gl.NONE, gl.COLOR_ATTACHMENT1]);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "should be no errors when reading from GL_COLOR_ATTACHMENT1 on fbo.");
+  testPixels(0, 0, fboColor1);
+
+  // switch to another fbo, read buffer is GL_COLOR_ATTACHMENT0, not GL_COLOR_ATTACHMENT1
+  var fb1 = gl.createFramebuffer();
+  gl.bindFramebuffer(gl.FRAMEBUFFER, fb1);
+  var buffer = gl.createRenderbuffer();
+  gl.bindRenderbuffer(gl.RENDERBUFFER, buffer);
+  gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, buffer);
+  gl.renderbufferStorage(gl.RENDERBUFFER, gl.RGBA8, canvas.width, canvas.height);
+
+  shouldBe('gl.getParameter(gl.READ_BUFFER)', 'gl.COLOR_ATTACHMENT0');
+  var fbo1Color = [255, 255, 255, 255];
+  gl.clearColor(fbo1Color[0] / 255, fbo1Color[1] / 255, fbo1Color[2] / 255, fbo1Color[3] / 255);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "should be no errors when reading from GL_COLOR_ATTACHMENT0 on another fbo.");
+  testPixels(0, 0, fbo1Color);
+
+  // switch from fbo to default fb, read buffer will switch to GL_BACK from GL_COLOR_ATTACHMENT0
+  gl.bindFramebuffer(gl.FRAMEBUFFER, null);
+  gl.deleteFramebuffer(fb);
+  gl.deleteRenderbuffer(colorbuffer);
+  gl.deleteRenderbuffer(colorbuffer1);
+  gl.deleteFramebuffer(fb1);
+  gl.deleteRenderbuffer(buffer);
+  gl.deleteTexture(tex);
+  shouldBe('gl.getParameter(gl.READ_BUFFER)', 'gl.BACK');
+  gl.copyTexImage2D(gl.TEXTURE_2D, 0, gl.RGBA, 0, 0, canvas.width, canvas.height, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR,
+      "should be no errors when reading from GL_BACK after switch to default fb from fbo.");
```

Comment:
The texture which was the target of the copyTexImage2D call was deleted four lines above, and was bound to the TEXTURE_2D target, so it's as though bindTexture(TEXTURE_2D, null) was called, and WebGL doesn't allow copyTexImage2D to work without a texture bound to the target texture unit. Therefore this call will produce an INVALID_OPERATION error. Per discussion above, I think it should be removed, and readPixels calls used everywhere in this test.